app-creator/src/widgets/ui-panel/ui-panel.ts

```diff
@@ -42,40 +76,99 @@ export class Panel extends LitElement {
   @property({ type: Object }) styles = {};
 
   /**
-   * Sets the flex direction of child widgets.
+   * Sets the flex layout of child widgets.
    * Options available are 'column' and 'row'.
-   * column direction will append widgets below the last child element.
-   * row direction will append widgets to the right of the last child element.
+   * column layout will append widgets below the last child element.
+   * row layout will append widgets to the right of the last child element.
    */
-  @property({ type: String }) direction = 'column';
+  @property({ type: String }) layout = 'column';
 
   /**
    * Adds a border and shadow to panel.
    */
   @property({ type: Boolean }) isRaised = false;
 
+  /**
+   * Contains an inner dropzone-widget.
+   */
+  @property({ type: Boolean }) hasDropzone = false;
+
+  /**
+   * Adds padding to panel.
+   */
+  @property({ type: Boolean }) padded = false;
+
+  /**
+   * Adds edit icon to panel.
+   */
+  @property({ type: Boolean }) editable = false;
+
   render() {
-    const { isRaised, styles } = this;
+    const { isRaised, editable, layout, padded, handleEditWidget } = this;
+
+    const editableMarkup = editable
+      ? html`
+          <div id="editable-view">
+            <iron-icon
+              class="edit-buttons"
+              icon="create"
+              @click=${handleEditWidget}
+            ></iron-icon>
+          </div>
+        `
+      : nothing;
+
     return html`
       <div
         id="container"
-        class="${classMap({ raised: isRaised })}"
-        style="${styleMap(styles)}"
+        class="${layout} ${isRaised ? 'raised' : ''} ${padded ? 'padded' : ''}"
       >
-        <dropzone-widget></dropzone-widget>
+        <slot class="${layout}"></slot>
+        ${editableMarkup}
       </div>
     `;
   }
 
-  getDirection() {
-    return this.direction;
+  /**
+   * Triggered when the edit icon is clicked. Stores a reference of the selected element in the store and
+   * displays a set of inputs for editing its attributes.
+   */
+  handleEditWidget() {
+    // Check if a widgetRef has been set.
+    store.dispatch(setEditingWidget(this));
+  }
+
+  setAttribute(key: string, value: string) {
+    switch (key) {
+      case 'layout':
+        this.layout = value;
+        return;
+      case 'isRaised':
+        this.isRaised = value === 'true';
+        return;
+      case 'hasDropzone':
+        this.hasDropzone = value === 'true';
+        return;
+      case 'padded':
+        this.padded = value === 'true';
+        return;
+    }
+
+    this.requestUpdate();
+  }
+
+  getlayout() {
+    return this.layout;
   }
 
   getStyle(): object {
     return this.styles;
   }
 
   setStyle(style: { [key: string]: string }) {
-    this.styles = style;
+    for (const attribute in style) {
```

Comment:
1. Objects are not iterable, so we cannot use `for...of`, but we can use `for...in` to iterate through enumerable properties (keys). We can also do something like `for ... of Object.keys(this.style);` 

2. We cannot overwrite the style property since it is read only. This is what I tried. Is there a different way that we can do it?
`this.style = { ...this.style, ...style };`.

Reply:
Ahh I see! Using `for..in` makes sense in the case then. :)